test/00_unit/unary_op_test.cpp

```diff
@@ -0,0 +1,262 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (C) 2023-2025 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <gtest/gtest.h>
+#include <hip/hip_runtime.h>
+#include <hiptensor/internal/hiptensor_utility.hpp>
+#include <hiptensor_unary_element_wise_operation.hpp>
+#include <iostream>
+
+using ck::tensor_operation::element_wise::HiptensorUnaryOp;
+// using HiptensorUnaryOp = ck::tensor_operation::element_wise::HiptensorUnaryOp;
+
+// Kernel function to add two arrays
+__global__ void unary_op_kernel(const float* x, float* y, int size, hiptensorOperator_t op_type)
+{
+    int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    if(idx < size)
+    {
+        HiptensorUnaryOp op(op_type);
+        op(y[idx], x[idx]);
+    }
+}
+
+float unaryOpOnDeviceTest(float x, hiptensorOperator_t op_type)
+{
+    const int array_size  = 64;
+    const int array_bytes = array_size * sizeof(float);
+
+    // Host arrays
+    float array_x[array_size] = {0.0F};
+    float array_y[array_size] = {0.0F};
+
+    array_x[0] = x;
```

Comment:
Should we initialize all values with x, and then compare all values instead of just the first?